ocaml/typing/mode.ml

```diff
@@ -349,22 +296,28 @@ module Lattices_mono = struct
         (** The constant morphism that always maps to the minimum *)
     | Const_max : 'a obj -> ('a, 'b, disallowed * 'd) morph
         (** The constant morphism that always maps to the maximum *)
-    | Proj :
-        ('a0, 'a1) Product.t obj * ('a0, 'a1, 'a) Product.axis
-        -> (('a0, 'a1) Product.t, 'a, 'l * 'r) morph
-        (** projection from product to an axis *)
-    | Max_with :
-        ('a0, 'a1, 'a) Product.axis
-        -> ('a, ('a0, 'a1) Product.t, disallowed * 'r) morph
-        (** Maps to maximum product except the given axis *)
-    | Min_with :
-        ('a0, 'a1, 'a) Product.axis
-        -> ('a, ('a0, 'a1) Product.t, 'l * disallowed) morph
-        (** Maps to minimum product except the given axis *)
+    | Proj_areality :
+        'a and_comonads obj
+        -> ('a and_comonads, 'a, 'l * 'r) morph
+        (** Project the areality component from the comonads *)
+    | Proj_linearity :
+        'a and_comonads obj
+        -> ('a and_comonads, Linearity.t, 'l * 'r) morph
+        (** Project the linearity component from the comonads *)
```

Comment:
`Proj` is splitted into two, which is less abstraction. This makes `maybe_compose` more complex and will get worse as we add more axes. I'm refactoring this and I think it can work.

Reply:
I have pushed the refactoring - see if you like it :smiley:

Reply:
I do! There's a few edits I want to try out, but I think this is the right move.

Reply:
My edits all ended in disaster. (I was trying to get rid of the `assert false` you added. But it was not to be.) Let's just go with this as is.